sympy/solvers/solveset.py

```diff
@@ -149,6 +149,32 @@ def _invert_real(f, g_ys, symbol):
         if g is not S.Zero:
             return _invert_real(h, imageset(Lambda(n, n - g), g_ys), symbol)
 
+        # if there are two terms with g_ys=0
+        # e.g. 3*f(x) + 2*g(x) -> f(x)/g(x) = -2/3,
+        # where f(x) and g(x) are power terms
+        if not g and not g_ys.args[0] and len(h.args) == 2 and \
```

Comment:
In a system like solvers where almost everything is heuristics, these conditions don't makes sense easily and would slowly get out of hand after multiple iterations of code changes over time. This is one of the reasons why `solve` is hard to understand.

To handle this situation, I would suggest to first document the what, why and how each and every condition and its proof of correctness, that would help in further development and extension. You also need to write meaningful functions wherever there is an independent piece/logic of code, which may or may not change to incorporate changes/addition of more conditions or situations to handle more equations.